### PR TITLE
API: Delete Document

### PR DESCRIPTION
**Pull request by Vidyaranya-Gavai-TTPL**
## 📋 Description

JIRA ID: 

Please provide a summary of the change and the motivation behind it. Include relevant context and details.

---

## ✅ Type of Change

- [ ] 🐞 **Bug fix** (non-breaking change which resolves an issue)
- [ ] ✨ **New feature** (non-breaking change which adds functionality)
- [ ] 🔥 **Breaking change** (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🛠 **Refactor** (change that is neither a fix nor a new feature)
- [ ] ⚙️ **Config change** (configuration file or build script updates)
- [ ] 📚 **Documentation** (updates to docs or readme)
- [ ] 🧪 **Tests** (adding new or updating existing tests)
- [ ] 🎨 **UI/UX** (changes that affect the user interface)
- [ ] 🚀 **Performance** (improves performance)
- [ ] 🧹 **Chore** (miscellaneous changes that don't modify src or test files)

---

## ℹ️ Additional Information

Please describe how the changes were tested, and include any relevant screenshots, logs, or other information that provides additional context.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new endpoint for deleting user documents, enhancing user control over their data.
	- Implemented authentication checks to ensure only authorized users can delete documents.

- **Bug Fixes**
	- Improved error handling for document deletion, including checks for user existence and document validity.

### CODE REVIEW

File: src/modules/users/users.controller.ts [hunk 1]
```diff
@@ -179,4 +182,26 @@ export class UserController {
...
+
+  @Delete('/delete-doc/:doc_id')
+  @UseGuards(AuthGuard)
+  @ApiOperation({ summary: 'Delete a document' })
+  @ApiBasicAuth('access-token')
+  @ApiResponse({ status: 200, description: 'Document deleted successfully' })
+  @ApiResponse({ status: 400, description: 'Document not found' })
+  @ApiResponse({ status: 401, description: 'Unauthorized' })
+  @ApiResponse({ status: 500, description: 'Internal server error' })
+  async deleteDoc(@Req() req: any, @Param('doc_id') doc_id: string) {
```
[the remark this answers is not in the document]
Comment (manojLondhe) [reply src/modules/users/users.controller.ts:194]: @Vidyaranya-Gavai-TTPL fix

File: src/modules/users/users.service.ts [hunk 2]
```diff
@@ -826,4 +826,66 @@ export class UserService {
...
+      where: { sso_id: sso_id },
+    });
+
+    if (!user)
+      return new ErrorResponse({
+        statusCode: HttpStatus.NOT_FOUND,
+        errorMessage: 'User with given sso_id not found',
+      });
+
+    const user_id = user.user_id;
```
[the remark this answers is not in the document]
Comment (manojLondhe) [reply src/modules/users/users.service.ts:842]: @Vidyaranya-Gavai-TTPL fix

File: src/modules/users/users.controller.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Vidyaranya-Gavai-TTPL) [reply src/modules/users/users.controller.ts:194]: @manojLondhe sir, If the type of req is set to Request, it gives error on next line. It says, 'Type Request does not contain property user'.

File: src/modules/users/users.service.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Vidyaranya-Gavai-TTPL) [reply src/modules/users/users.service.ts:842]: @manojLondhe sir, there is no 'keycloak_id' property in users entity

**Pull request merged**